Uplift/Views/Footers/GymCellFooter.swift

```diff
@@ -0,0 +1,151 @@
+//
+//  GymCellFooter.swift
+//  Uplift
+//
+//  Created by Elvis Marcelo on 3/22/23.
+//  Copyright © 2023 Cornell AppDev. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+class GymCellFooter: UIView {
+    private let hoursLabel = UILabel()
+    private let locationNameLabel = UILabel()
+    private let statusLabel = UILabel()
+    private let capacityStatusLabel = UILabel()
+    private let capacityCountLabel = UILabel()
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        backgroundColor = .white
+        setupViews()
+        setupConstraints()
+    }
+
+    func configure(for gym: Gym) {
+        locationNameLabel.text = gym.name
+
+        //set gym status
+        let changingSoon = gym.isStatusChangingSoon()
+
+        if(gym.isOpen) {
+            statusLabel.textColor = changingSoon ? .accentOrange : .accentOpen
+        } else {
+            statusLabel.textColor = changingSoon ? .accentOrange : .accentClosed
+        }
+
+        statusLabel.text = gym.isOpen ? ClientStrings.CommonStrings.open : ClientStrings.CommonStrings.closed
+
+        // Set gym hours
+        hoursLabel.text = getHoursString(from: gym)
+
+        capacityStatusLabel.text = "Cramped"
+        capacityCountLabel.text = "120/140"
+    }
+
+    private func getHoursString(from gym: Gym) -> String {
+        let now = Date()
+        let isOpen = gym.isOpen
+
+        let gymHoursToday = gym.gymHoursToday
+        let gymHoursTomorrow = gym.gymHours[now.getIntegerDayOfWeekTomorrow()]
+
+        let format: String
+
+        if now > gymHoursToday.closeTime {
+            format = gymHoursTomorrow.openTime.getHourFormat()
+            if gym.closedTomorrow {
+                return ClientStrings.CommonStrings.closed
+            } else {
+                return ClientStrings.Home.gymDetailCellOpensAt + gymHoursTomorrow.openTime.getStringOfDatetime(format: format)
+            }
+        } else if !isOpen {
+            format = gymHoursToday.openTime.getHourFormat()
+            return ClientStrings.Home.gymDetailCellOpensAt + (gymHoursToday.openTime.getStringOfDatetime(format: format))
+        } else {
+            format = gymHoursToday.closeTime.getHourFormat()
+            let openTime = gymHoursToday.openTime.getStringOfDatetime(format: format)
+            let closeTime = gymHoursToday.closeTime.getStringOfDatetime(format: format)
+
+            return ClientStrings.Home.gymDetailCellClosesAt + closeTime
+        }
```

Comment:
This seems a little unintuitive to me/hear to hear. I'll differ judgment to @ReadeP if it actually needs to/should be reworked.

At the least, I think you should comment this to make it very clear what is happening.